core/src/main/java/com/linecorp/armeria/client/retry/RetryingClient.java

```diff
@@ -44,25 +41,27 @@
     private static final AttributeKey<Long> RESPONSE_TIMEOUT_DEADLINE_NANOS =
             AttributeKey.valueOf("RESPONSE_TIMEOUT_DEADLINE_NANOS");
 
-    private final Supplier<? extends Backoff> backoffSupplier;
+    private static final AttributeKey<RetryStrategyState> LAST_RETRY_STRATEGY_STATE =
+            AttributeKey.valueOf("LAST_RETRY_STRATEGY_STATE");
```

Comment:
- `AttributeKey.valueOf(RetryingClient.class, "LAST_RETRY_STRATEGY_STATE");`
- `AttributeKey.valueOf(RetryingClient.class, "RESPONSE_TIMEOUT_DEADLINE_NANOS");`

We may want to merge these two attributes into one such as `RetryingClientState`.